src/frontend/utils/parse_util.py

```diff
@@ -0,0 +1,10 @@
+from django.http import Http404
+
+def parse_int_or_404(x):
+    if x:
+        try:
+            parsed = int(x)
+            return parsed
+        except ValueError:
+            raise Http404("Invalid argument")
```

Comment:
Is 404 the right error for this. I would have said 400.

Reply:
Isn't 404 more consistent with what happens when in general you access a page that doesn't exist? Like katagotraining.org/networks/network-that-does-not-exist/?